### PR TITLE
New notifier to report counter metrics and events to DataDog

### PR DESCRIPTION
**Pull request by volodkevych**

### CODE REVIEW

File: Nimator/Settings/DataDogSettings.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            return new DataDogSettings
+            {
+                Prefix = "Nimator",
+                StatsdServerName = "127.0.0.1",
+                StatsdPort = 8125,
+                Threshold = NotificationLevel.Warning
+            };
+        }
+
+        private void CongifureStatsd()
```
Comment (douwequerty) [inline Nimator/Settings/DataDogSettings.cs]: ConfigureStats

File: Nimator/Notifiers/DataDogNotifier.cs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Nimator.Settings;
+using Nimator.Util;
+using StatsdClient;
+
+namespace Nimator.Notifiers
+{
+    class DataDogNotifier : INotifier
```
Comment (douwequerty) [inline Nimator/Notifiers/DataDogNotifier.cs]: no public ?

File: Nimator/Notifiers/DataDogNotifier.cs [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    {
+        private DataDogSettings settings;
```
Comment (douwequerty) [inline Nimator/Notifiers/DataDogNotifier.cs]: readonly ?

File: Nimator/Notifiers/DataDogNotifier.cs [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+                    GetMessage(checkResult.RenderPlainText()));
+                }
+            }
+        }
+
+        private void AddStatsdMertics(NotificationLevel level, string layerName, string checkName, string message)
+        {
+            DogStatsd.Increment(statName: checkName);
+            DogStatsd.Event(title: checkName,
+                text: message,
```
Comment (douwequerty) [inline Nimator/Notifiers/DataDogNotifier.cs]: Need all the names here ?

File: Nimator/Notifiers/DataDogNotifier.cs [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+            if (!result.LayerResults.Any())
+            {
+                AddStatsdMertics(result.Level,
+                    result.GetFirstFailedLayerName(),
+                    result.GetFirstFailedCheckName(),
+                    GetMessage(result.Message, result.RenderPlainText(settings.Threshold)));
+            }
+
+            foreach(var layerResult in result.LayerResults.Where(l => l?.Level >= settings.Threshold))
```
Comment (douwequerty) [inline Nimator/Notifiers/DataDogNotifier.cs]: Could we separate the where on a line and foreach over a variable or put the body of the foreach in a private method and go for the select ? I think that would be more readable

File: Nimator/Settings/DataDogSettings.cs (hunk 1 above)
Comment (volodkevych) [reply Nimator/Settings/DataDogSettings.cs]: This method is configuring DogStatsd, which is part of **Statsd**Client. That's why I called this method corresponding to the name of dependency. Another option could be ConfigureDogStatsd.

File: Nimator/Settings/DataDogSettings.cs [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+namespace Nimator.Settings
+{
+    /// <summary>
+    /// Settings for creating a <see cref="INotifier"/> that will publish to DataDog: <see href="https://www.datadoghq.com/">datadoghq.com</see>
+    /// </summary>
+    public class DataDogSettings : NotifierSettings
+    {
+        /// <summary>
+        ///
```
Comment (douwequerty) [inline Nimator/Settings/DataDogSettings.cs]: Some decription would be nice

**Comment by volodkevych** [comment]
@douwequerty I handled your review comments. Let me know if I can merge now.

**Comment by douwequerty** [comment]
Sure go ahead and merge it

**Pull request merged**